### PR TITLE
fix: read version and domain from path instead of filename for go

### PR DESCRIPTION
**Pull request by tiwarishubham635**
# Fixes #

Fetch version from open api spec paths instead of file name. 
**A breaking change for go.**

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] Run `make test-docker`
- [ ] Verify affected language:
    - [ ] Generate [twilio-go](https://github.com/twilio/twilio-go) from our [OpenAPI specification](https://github.com/twilio/twilio-oai) using the [build_twilio_go.py](./examples/build_twilio_go.py) using `python examples/build_twilio_go.py path/to/twilio-oai/spec/yaml path/to/twilio-go` and inspect the diff
    - [ ] Run `make test` in `twilio-go`
    - [ ] Create a pull request in `twilio-go`
    - [ ] Provide a link below to the pull request
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-oai-generator/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please create a GitHub Issue in this repository.

### CODE REVIEW

File: src/main/java/com/twilio/oai/AbstractTwilioGoGenerator.java [hunk 1]
```diff
@@ -78,6 +78,15 @@ public void processOpenAPI(final OpenAPI openAPI) {
                 param.addExtension(ACCOUNT_SID_VEND_EXT, true);
             });
 
+        String domain = StringHelper.toSnakeCase(twilioCodegen.getDomainFromOpenAPI(openAPI));
+        int lastIndex = domain.lastIndexOf("_api");
```
Comment (sbansla) [inline src/main/java/com/twilio/oai/AbstractTwilioGoGenerator.java:82]: add a comment here that we are skipping _api for backward compatibility of folder names.

**Review by sbansla — CHANGES_REQUESTED** [review]
add a comment here that we are skipping _api for backward compatibility of folder names.
Give example that frontline-api server won't be converted to frontline_api folder name but frontline for backward compatibility.

File: src/main/java/com/twilio/oai/AbstractTwilioGoGenerator.java (hunk 1 above)
Comment (tiwarishubham635) [reply src/main/java/com/twilio/oai/AbstractTwilioGoGenerator.java:82]: Got it

Comment (tiwarishubham635) [reply src/main/java/com/twilio/oai/AbstractTwilioGoGenerator.java:87]: Done

**Review by sbansla — APPROVED** [review]

**Pull request merged**